aquadoggo/src/replication/manager.rs

```diff
@@ -452,11 +587,350 @@ mod tests {
             let result = manager.handle_message(&PEER_ID_REMOTE, &message).await;
             assert!(matches!(
                 result,
-                Err(ReplicationError::DuplicateInboundRequest(1))
+                Err(ReplicationError::DuplicateSession(err)) if err == DuplicateSessionRequestError::InboundExistingTargetSet(target_set_2)
             ));
         })
     }
 
+    //  PEER A                                 PEER B
+    //
+    //  SyncRequest(0, 0, ["A"])────────────────────►
+    //
+    //  ◄─────────────────── SyncRequest(0, 0, ["B"])
+    //
+    //  ========== PEER A REQUEST DROPPED ===========
+    //
+    //  0 Have([..]) ───────────────────────────────►
+    //  0 SyncDone(false) ──────────────────────────►
```

Comment:
Think the diagrams need to look like that (no `SyncDone` being moved to the end). Our messages are FIFO-based, messages arrive in the order they were sent, one by one, nothing gets re-ordered. The diagram before suggested that the first `SyncDone` message gets ordered last, not sure if that was indended though.

Reply:
It was actually intentional to do it as it was, the reason being that I wanted to demonstrate the order messages were processed on each node, as well as when theyre sent. Think this is most useful for understanding the state changes. Peer B processes Peer A's `Have` message, generates and sends  a `Have` and `SyncDone`, _then_ processes the `SyncDone`.

Reply:
Cool, I understand now!